go/adapter/honeycomb/adapter.go

```diff
@@ -0,0 +1,134 @@
+package honeycomb
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+
+	observe "github.com/dylibso/observe-sdk/go"
+	otel "github.com/dylibso/observe-sdk/go/adapter/otel_formatter"
+	trace "go.opentelemetry.io/proto/otlp/trace/v1"
+	proto "google.golang.org/protobuf/proto"
+)
+
+type HoneycombConfig struct {
+	ApiKey             string
+	Dataset            string
+	EmitTracesInterval uint32
+	TraceBatchMax      uint32
+	Host               string
+}
+
+type HoneycombAdapter struct {
+	*observe.AdapterBase
+	Config *HoneycombConfig
+}
+
+func NewHoneycombAdapter(config *HoneycombConfig) *HoneycombAdapter {
+	base := observe.NewAdapterBase(1, 0)
+	adapter := &HoneycombAdapter{
+		AdapterBase: &base,
+		Config:      config,
+	}
+
+	adapter.AdapterBase.SetFlusher(adapter)
+
+	return adapter
+}
+
+func (h *HoneycombAdapter) Start() {
+	h.AdapterBase.Start(h)
+}
+
+func (h *HoneycombAdapter) HandleTraceEvent(te observe.TraceEvent) {
+	h.AdapterBase.HandleTraceEvent(te)
+}
+
+func (h *HoneycombAdapter) Flush(evts []observe.TraceEvent) error {
+	for _, te := range evts {
+		traceId := te.TelemetryId.ToHex16()
+
+		var allSpans []*trace.Span
+		for _, e := range te.Events {
+			switch event := e.(type) {
+			case observe.CallEvent: // TODO: consider renaming to FunctionCall for consistency across Rust & JS
+				spans := h.makeCallSpans(event, nil, traceId)
+				if len(spans) > 0 {
+					allSpans = append(allSpans, spans...)
+				}
+			case observe.MemoryGrowEvent:
+				log.Println("MemoryGrowEvent should be attached to a span")
+			case observe.CustomEvent:
+				log.Println("Otel adapter does not respect custom events")
+			}
+		}
+
+		if len(allSpans) <= 1 {
+			log.Println("No spans built for honeycomb")
+			return nil
+		}
+
+		t := otel.NewTrace(traceId, h.Config.Dataset, allSpans)
+		data, err := proto.Marshal(t.TracesData)
+		if err != nil {
+			log.Println("failed to marshal TracesData:", err)
+			return nil
+		}
+
+		url, err := url.JoinPath(h.Config.Host, "v1", "traces")
+		if err != nil {
+			log.Println("failed to create honeycomb endpoint url:", err)
+			return nil
+		}
+
+		client := http.Client{}
```

Comment:
we probably don't do this in the other Go adapters (but should).. but this http client has _no_ timeout by default - so it will just hang open forever. not sure exactly what the right duration is, but I think if we start with something small, like 2s, it would probably be a good idea

Reply:
Good call, adding in the timeout. The other adapters are just using `http.Post` currently which I don't see a way to specify a timeout.  Maybe we can add the timeout to the other adapters in a follow up pass and see if this logic can be abstracted into the base adapter